src/app/repositories/buckets/dataset_bucket_repository.py

```diff
@@ -0,0 +1,24 @@
+from google.cloud import storage
+from logging_config import logging
+from models.dataset_models import RawDatasetWithMetadata
+from services.shared.bucket_operations_service import BucketOperationsService
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetBucketRepository:
+    def __init__(self, bucket_name: str):
+        self.storage_client = storage.Client()
+        self.bucket = self.storage_client.bucket(bucket_name)
+
+    def get_bucket_file_as_json(self, filename: str) -> RawDatasetWithMetadata:
+        """
+        Queries google bucket for file with a specific name and returns it as json.
+
+        Parameters:
+        filename (str): name of file being queried.
+
+        Returns:
+        RawDatasetWithMetadata: raw dataset from the bucket file as json.
+        """
+        return BucketOperationsService.get_bucket_file_as_json(filename, self.bucket)
```

Comment:
It seems a bit strange that and additional service class has to be called to perform the file retrieval process, as normally it will be assumed that the repository layer will perform the retrieval task itself, in consistent with how the firebase repository classes works, instead of performing it at the service layer.
But as the file retrieval process should be consistent among schema and dataset, I recommend we can try using a parent class eg. BucketRepository to handle the get_bucket_file_as_json process, while both DatasetBucketRepository and SchemaBucketRepository inherit it to make parameters adjustment

Reply:
Yeah wasn't 100% happy with this part of the refactor, I like your suggestion and will implement it now.

Reply:
I've implemented your suggestion, let me know what you think